File: promotion/api-gateway/src/main/java/com/study/apigateway/controller/FallbackController.java
package com.study.apigateway.controller;

import java.util.Map;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@Slf4j
@RestController
@RequestMapping("/fallback")
public class FallbackController {
  @GetMapping("/users")
  public Mono<Map<String, Object>> userFallback() {
    return Mono.just(Map.of("status", "down"));
  }
}

File: promotion-kotlin/api-gateway-kotlin/src/main/kotlin/com/study/apigatewaykotlin/controller/FallbackController.kt
package com.study.apigatewaykotlin.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/fallback")
class FallbackController {

    @GetMapping("/users")
    fun userFallback(): Mono<Map<String, *>> {
        return Mono.just(mapOf(Pair("status", "down")))
    }
}
